File: accommodation_service/src/main/java/huy/project/accommodation_service/infrastructure/repository/mapper/LanguageMapper.java
package huy.project.accommodation_service.infrastructure.repository.mapper;

import huy.project.accommodation_service.core.domain.entity.LanguageEntity;
import huy.project.accommodation_service.infrastructure.repository.model.LanguageModel;
import org.mapstruct.Mapper;
import org.mapstruct.factory.Mappers;

@Mapper
public abstract class LanguageMapper {
    public static final LanguageMapper INSTANCE = Mappers.getMapper(LanguageMapper.class);

    public abstract LanguageEntity toEntity(LanguageModel language);

    public abstract LanguageModel toModel(LanguageEntity language);
}

File: location_service/LocationService.Infrastructure/Repository/Mapper/LanguageMapper.cs
namespace LocationService.Infrastructure.Repository.Mapper
{
    using LocationService.Core.Domain.Entity;
    using LocationService.Infrastructure.Repository.Model;

    public class LanguageMapper
    {
        public static LanguageEntity ToEntity(LanguageModel model)
        {
            if (model == null) return null!;
            return new LanguageEntity
            {
                Id = model.Id,
                Name = model.Name,
                Code = model.Code
            };
        }

        public static LanguageModel ToModel(LanguageEntity entity)
        {
            if (entity == null) return null!;
            return new LanguageModel
            {
                Id = entity.Id,
                Name = entity.Name,
                Code = entity.Code
            };
        }
    }
}